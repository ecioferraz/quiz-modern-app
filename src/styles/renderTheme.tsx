import { render, RenderResult } from '@testing-library/react';
import { ReactNode } from 'react';
import { ThemeProvider } from 'styled-components';
import { theme } from './theme';

export default function renderTheme(children: ReactNode): RenderResult {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
}
