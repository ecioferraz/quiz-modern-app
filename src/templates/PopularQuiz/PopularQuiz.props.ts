type PopularQuizProps = {
  background: string;
  category: string;
  iconSrc: string;
};

export default PopularQuizProps;
