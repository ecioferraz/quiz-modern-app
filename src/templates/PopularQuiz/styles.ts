import styled, { css } from 'styled-components';

export const Container = styled.section`
  ${({ theme }) => css`
    h1 {
      color: ${theme.font.color.tertiaryColor};
      font-weight: ${theme.font.weight.semiBold};
      margin: 3.2rem 0 1.6rem 0;
    }

    a {
      color: ${theme.font.color.heroColor};
      text-decoration: none;
    }
  `}
`;
