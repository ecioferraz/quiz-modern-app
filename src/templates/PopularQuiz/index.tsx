import TextCard from '../../components/TextCard';
import CategoryCard from '../CategoryCard';
import CategoryCardProps from '../CategoryCard/CategoryCard.props';
import * as Styled from './styles';

export default function PopularQuiz() {
  const popularQuizCards: CategoryCardProps[] = [
    {
      background: '#FFE3BE',
      description: '10 Questions',
      iconSrc: 'assets/icons/desktop.svg',
      title: 'Technology',
    },
    {
      background: '#EAEEFF',
      description: '10 Questions',
      iconSrc: 'assets/icons/ball.svg',
      title: 'Sport',
    },
    {
      background: '#F6E5E5',
      description: '10 Questions',
      iconSrc: 'assets/icons/music.svg',
      title: 'Music',
    },
  ];

  return (
    <Styled.Container>
      <TextCard size="small">Popular Quiz</TextCard>
      {popularQuizCards.map(({ background, description, iconSrc, title }) => (
        <CategoryCard
          background={background}
          description={description}
          iconSrc={iconSrc}
          key={title}
          rating
          title={title}
        />
      ))}
    </Styled.Container>
  );
}
