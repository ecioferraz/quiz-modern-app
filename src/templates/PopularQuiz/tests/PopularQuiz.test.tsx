import { describe, expect, it } from 'vitest';
import renderTheme from '../../../styles/renderTheme';
import PopularQuiz from '..';

describe('<PopularQuiz />', () => {
  it('should render <PopularQuiz />', () => {
    const { container } = renderTheme(<PopularQuiz />);

    expect(container).toMatchSnapshot();
  });
});
