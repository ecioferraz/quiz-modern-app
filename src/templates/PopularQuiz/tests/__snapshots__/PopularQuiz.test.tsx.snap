// Vitest Snapshot v1

exports[`<PopularQuiz /> > should render <PopularQuiz /> 1`] = `
.c1 {
  font-size: 1.6rem;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #FFE3BE;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c8 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #EAEEFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c9 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #F6E5E5;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #FFFFFF;
  box-shadow: 10px 24px 54px rgba(51,51,51,0.03);
  border-radius: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 1.6rem 0;
}

.c2 > * {
  margin: 1.6rem;
  margin-right: 1.9rem;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1.6rem;
}

.c3 > section:has(img) {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 10px;
  box-shadow: 10px 24px 54px rgba(51,51,51,0.05);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 50px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 50px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.8rem;
}

.c5 h2,
.c5 p {
  margin: 0;
}

.c5 h2 {
  font-size: 1.4rem;
  font-weight: 600;
}

.c5 p {
  color: #999999;
  font-family: "Nunito",sans-serif;
  font-size: 1.2rem;
  font-weight: 400;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.6rem;
}

.c6 p {
  font-size: 1.4rem;
}

.c0 h1 {
  color: #999999;
  font-weight: 600;
  margin: 3.2rem 0 1.6rem 0;
}

.c0 a {
  color: #C4C4C4;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div>
  <section
    class="c0"
  >
    <h1
      class="c1"
    >
      Popular Quiz
    </h1>
    <section
      class="c2"
    >
      <section
        class="c3"
      >
        <section
          class="c4"
        >
          <img
            alt="Technology icon"
            src="assets/icons/desktop.svg"
          />
        </section>
        <section
          class="c5"
        >
          <h2
            class="c1"
          >
            Technology
          </h2>
          <p
            class="c1"
          >
            10 Questions
          </p>
        </section>
      </section>
      <section
        class="c6"
      >
        <section
          class="c7"
        >
          <img
            alt="Rating star"
            src="assets/icons/star.svg"
          />
        </section>
        <p
          class="c1"
        >
          4.8
        </p>
      </section>
    </section>
    <section
      class="c2"
    >
      <section
        class="c3"
      >
        <section
          class="c8"
        >
          <img
            alt="Sport icon"
            src="assets/icons/ball.svg"
          />
        </section>
        <section
          class="c5"
        >
          <h2
            class="c1"
          >
            Sport
          </h2>
          <p
            class="c1"
          >
            10 Questions
          </p>
        </section>
      </section>
      <section
        class="c6"
      >
        <section
          class="c7"
        >
          <img
            alt="Rating star"
            src="assets/icons/star.svg"
          />
        </section>
        <p
          class="c1"
        >
          4.8
        </p>
      </section>
    </section>
    <section
      class="c2"
    >
      <section
        class="c3"
      >
        <section
          class="c9"
        >
          <img
            alt="Music icon"
            src="assets/icons/music.svg"
          />
        </section>
        <section
          class="c5"
        >
          <h2
            class="c1"
          >
            Music
          </h2>
          <p
            class="c1"
          >
            10 Questions
          </p>
        </section>
      </section>
      <section
        class="c6"
      >
        <section
          class="c7"
        >
          <img
            alt="Rating star"
            src="assets/icons/star.svg"
          />
        </section>
        <p
          class="c1"
        >
          4.8
        </p>
      </section>
    </section>
  </section>
</div>
`;
